react_django/django-src/django_app/api.py

```diff
@@ -28,12 +32,40 @@ def create_user(request):
 def log_in(request):
     if request.META['REQUEST_METHOD'] == 'POST':
         request_body = json.loads(request.body.decode('ascii'))
-        user = UserProfile.objects.filter(username=request_body['username'])[0]
-        return JsonResponse({"logged in": str(verify_password(request_body['password'], user.pwd))})
+        user = UserProfile.objects.filter(username=request_body['username'])
+        if(len(user) > 0):
+          user = user[0]
+          return JsonResponse({"logged in": str(verify_password(request_body['password'], user.pwd))}, status=202)
+        else:
+          return JsonResponse({"Login failed": 0}, status=400)
+
+def update_user(request):
+    if request.META['REQUEST_METHOD'] == 'PUT':
+      request_body = json.loads(request.body.decode('ascii'))
+      user = UserProfile.objects.filter(username=request_body['username'])
+      if(len(user) > 0):
+        user = user[0]
+        user.pwd = encrypt_password(request_body['password'])
+        try:
+          user.save()
+        except IntegrityError as e:
+          return JsonResponse({"Error":0},status=400)
+        return JsonResponse({"Password changed": str(verify_password(request_body['password'], user.pwd))}, status=202)
+      else:
+        return JsonResponse({"Error": 0}, status=400)
+
+
+
+
 
 
 def delete_user(request):
     if request.META['REQUEST_METHOD'] == 'DELETE':
         request_body = json.loads(request.body.decode('ascii'))
-        user = UserProfile.objects.filter(username=request_body['username'])[0].delete()
```

Comment:
So by accessing UserProfile (and other models) through their methods in this context, the changes are ultimately propagated to the DB, correct?

Maybe something like a comment header at the top of every file that explains what that file does in the context of the app... That may be something I can do for you all.